frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -65,7 +65,7 @@ export const generateClientKeySecret = (clientKey: string) => ({
   },
 });
 
-export const createKmsResources = (kms: KMSConfig, update = false, previousData?: any) => {
+export const createKmsResources = (kms: KMSConfig, update = false, createOcsKms = true) => {
```

Comment:
```suggestion
export const createKmsResources = (kms: KMSConfig, update = false, createOcsKms = true) => {
```
export const createKmsResources = (kms: KMSConfig, update = false, encryptionType) => {

encryptionType is an enum and members are CLUSTER_WIDE, STORAGE_CLASS_WIDE

Reply:
Also, you dont need an update flag, you can use encryptionType for update